File: src/main/java/org/project/heredoggy/user/walk/walkRecord/dto/WalkRecordResponseDTO.java
package org.project.heredoggy.user.walk.walkRecord.dto;

import lombok.*;
import org.project.heredoggy.domain.postgresql.walk.walkRecord.WalkRecordStatus;

import java.time.LocalDateTime;
import java.util.List;

@Getter
@Setter
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class WalkRecordResponseDTO {
    private Long id;
    private Long reservationId;
    private Long walkRouteId;
    private Double actualDistance;
    private Integer actualDuration;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private List<WalkRecordPointDTO> actualPath;
    private WalkRecordStatus status;
    private String thumbnailUrl;
}

File: src/main/frontend/lib/models/walk_record_response_dto.dart
import 'package:flutter/foundation.dart';
import 'walk_record_point_dto.dart';

class WalkRecordResponseDTO {
  final int id;
  final int reservationId;
  final int walkRouteId;
  final double? actualDistance;
  final int? actualDuration;
  final String? startTime;
  final String? endTime;
  final List<WalkRecordPointDTO>? actualPath;
  final String status;
  final String? thumbnailUrl;

  WalkRecordResponseDTO({
    required this.id,
    required this.reservationId,
    required this.walkRouteId,
    this.actualDistance,
    this.actualDuration,
    this.startTime,
    this.endTime,
    this.actualPath,
    required this.status,
    this.thumbnailUrl,
  });

  factory WalkRecordResponseDTO.fromJson(Map<String, dynamic> json) {
    return WalkRecordResponseDTO(
      id: json['id'] as int,
      reservationId: json['reservationId'] as int,
      walkRouteId: json['walkRouteId'] as int,
      actualDistance: json['actualDistance'] as double?,
      actualDuration: json['actualDuration'] as int?,
      startTime: json['startTime'] as String?,
      endTime: json['endTime'] as String?,
      actualPath: json['actualPath'] != null
          ? List<WalkRecordPointDTO>.from(
              (json['actualPath'] as List).map(
                (x) => WalkRecordPointDTO.fromJson(x as Map<String, dynamic>),
              ),
            )
          : null,
      status: json['status'] as String,
      thumbnailUrl: json['thumbnailUrl'] as String?,
    );
  }
} 